src/Analyzers/Core/Analyzers/Helpers/DiagnosticHelper.cs

```diff
@@ -48,17 +63,12 @@ internal static class DiagnosticHelper
                 throw new ArgumentNullException(nameof(descriptor));
             }
 
-            LocalizableString message;
-            if (messageArgs == null || messageArgs.Length == 0)
-            {
-                message = descriptor.MessageFormat;
-            }
-            else
+            if (effectiveSeverity == ReportDiagnostic.Suppress)
             {
-                message = new LocalizableStringWithArguments(descriptor.MessageFormat, messageArgs);
+                return s_suppressedDiagnostic;
             }
 
-            return CreateWithMessage(descriptor, location, effectiveSeverity, additionalLocations, properties, message);
+            return Diagnostic.Create(descriptor, location, effectiveSeverity.ToDiagnosticSeverity() ?? descriptor.DefaultSeverity, additionalLocations, properties, messageArgs);
```

Comment:
The public compiler API doesn't have any API that accepts message format and message arguments along with isSuppressed flag.

The only API that takes isSuppressed takes a whole message. I'm using a static diagnostic that should be discarded by the compiler instead of IsSuppressed.